compaction/compaction_manager.cc

```diff
@@ -1034,8 +1034,14 @@ void compaction_manager::register_metrics() {
 }
 
 void compaction_manager::enable() {
-    SCYLLA_ASSERT(_state == state::none || _state == state::disabled);
-    _state = state::enabled;
+    SCYLLA_ASSERT(_state == state::none || _state == state::running);
+
+    if (_disabled_state_count > 0 && --_disabled_state_count > 0) {
```

Comment:
Protect against underflow with on_internal_error_noexcept

Reply:
Already protected with the first `_disabled_state_count > 0` check. The second check is to print a debug log.